code/xcvm/lib/core/src/gateway/config.rs

```diff
@@ -0,0 +1,217 @@
+pub struct IbcEnabled {
+	pub channels: Option<IbcChannels>,
+}
+
+/// we need both, so we can unwrap
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
+pub struct IcsPair {
+	pub source: ChannelId,
+	pub sink: ChannelId,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
+pub struct OtherNetworkItem {
+	pub ics_20: Option<IcsPair>,
+	/// default timeout to use for direct send
+	pub counterparty_timeout: IbcTimeout,
+	/// if there is custom IBC channel opened
+	pub xcvm_channel: Option<ChannelInfo>,
+}
+
+/// cross cross chain routing requires a lot of configuration,
+/// about chain executing this contract,
+/// about connectivity to and of other chains (even if not connected directly)
+/// and about assets and services on these chains
+/// (in future block hooks and some set of host extensions/precompiles would help to get some info
+/// automatically)
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
+pub enum ConfigSubMsg {
+	/// Permissioned message (gov or admin) to force set information about network contract is
+	/// executed. Network can be any network or this network (so it overrides some this network
+	/// parameters too)
+	ForceNetwork(NetworkItem),
+	/// Sets network to network connectivity/routing information
+	ForceNetworkToNetwork(ForceNetworkToNetworkMsg),
+
+	/// Permissioned message (gov or admin) to force set asset information.
+	ForceAsset(AssetItem),
+
+	/// Message sent by an admin to remove an asset from registry.
+	ForceRemoveAsset { asset_id: AssetId },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
+pub struct InstantiateMsg(pub HereItem);
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
+pub struct HereItem {
+	/// Network ID of this network
+	pub id: NetworkId,
+	/// The admin which is allowed to update the bridge list.
+	pub admin: Addr,
+}
+
+/// when message is sent to other side, we should identify receiver of some kind
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
+pub enum GatewayId {
+	CosmWasm {
+		contract: Addr,
+		/// Address of the XCVM interpreter contract code
+		interpreter_code_id: u64,
+		/// admin of everything
+		admin: Addr,
+	},
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
+
+pub struct AssetItem {
+	pub id: AssetId,
```

Comment:
Gonna be honest, I’d prefer `asset_id` name for this field.  Yes, it is a bit redundant but it introduces consistency throughout code where asset id is usually called `asset_id`.  It also helps in cases where types are inferred and it’s not obvious that structure one deals with is an `AssetItem`.  Same about `network_id` for network id in `NetworkItem` and `HereItem` actually.